src/gp_doctest.cpp

```diff
@@ -1562,3 +1562,147 @@ TEST_CASE("NNI Engine: NNI Likelihoods") {
     nni_count++;
   }
 }
+
+// Build set all valid node subsplits present in a complete SubsplitDAG.
+void CompleteDAGSubsplits(std::set<Bitset>& subsplits, size_t remaining,
+                          std::string assignment = "",
+                          bool first_left_subsplit = false) {
+  // If all bit positions have been assigned, generate clades.
+  if (remaining == 0) {
+    std::string left_clade = "";
+    std::string right_clade = "";
+    for (size_t i = 0; i < assignment.size(); i++) {
+      if (assignment[i] == '0') {
+        left_clade += "0";
+        right_clade += "0";
+      } else if (assignment[i] == '1') {
+        left_clade += "1";
+        right_clade += "0";
+      } else if (assignment[i] == '2') {
+        left_clade += "0";
+        right_clade += "1";
+      }
+    }
+    Bitset left_subsplit = Bitset(left_clade);
+    Bitset right_subsplit = Bitset(right_clade);
+    Bitset subsplit = Bitset::Subsplit(left_clade, right_clade);
+    // Only illegal bitsets: one clade is empty and the other clade is not a singleton
+    // (leaf) or all (universal ancestor).
+    if (left_subsplit.None() || right_subsplit.None()) {
+      if (!(left_subsplit.IsSingleton() || right_subsplit.IsSingleton())) {
+        if (!(left_subsplit.All() || right_subsplit.All())) {
+          return;
+        }
+      }
+    }
+    subsplits.insert(subsplit);
+    return;
+  }
+  // Otherwise, recurse on all possible assignments for next bit position.
+  // Outgroup.
+  CompleteDAGSubsplits(subsplits, remaining - 1, assignment + "0", first_left_subsplit);
+  // Left Subsplit.
+  CompleteDAGSubsplits(subsplits, remaining - 1, assignment + "1", true);
+  // Right Subsplit (the first non-outgroup must be assigned to the Left Subpslit).
+  if (first_left_subsplit) {
+    CompleteDAGSubsplits(subsplits, remaining - 1, assignment + "2", true);
+  }
+}
+// Build set all valid edge PCSPs present in a complete SubsplitDAG.
```

Comment:
I think we could give it an arbitrary collection of bitsets, no?

Reply:
Fixed comment.